metaflow/plugins/docker/fast_bakery.py

```diff
@@ -0,0 +1,172 @@
+import hashlib
+import json
+import requests
+
+from metaflow.exception import MetaflowException
+from metaflow.metaflow_config import (
+    FAST_BAKERY_TYPE,
+    FAST_BAKERY_URL,
+    FAST_BAKERY_AUTH,
+    get_pinned_conda_libs,
+)
+
+BAKERY_METAFILE = ".imagebakery-cache"
+
+
+class FastBakeryException(MetaflowException):
```

Comment:
We can throw a generic exception (not MetaflowException) here which can be caught by docker_environment and wrapped into a MetaflowException